extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +336,87 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    const TIME_WAIT = 500;
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
+    this.debouncedBoundOnSession_ = debounce(
+      this.root.ampdoc.win,
+      this.boundOnSession_,
+      TIME_WAIT
+    );
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(BrowserEventType).forEach((key) => {
+      root.removeEventListener(BrowserEventType[key], this.boundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.observables_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    userAssert(
+      isExperimentOn(this.root.ampdoc.win, 'custom-browser-event-tracker'),
+      'expected global "custom-browser-event-tracker" experiment to be enabled'
+    );
```

Comment:
These experiment names are global, and it's not clear that this is about analytics. Let's rename.

```suggestion
    userAssert(
      isExperimentOn(this.root.ampdoc.win, 'analytics-browser-events'),
      'expected global "analytics-browser-events" experiment to be enabled'
    );
```